airflow/models.py

```diff
@@ -2000,18 +2000,28 @@ def __cmp__(self, other):
 
     def pickle(self, main_session=None):
         session = main_session or settings.Session()
-        dag = session.query(
-            DagModel).filter(DAG.dag_id == self.dag_id).first()
+        orm_dag = session.query(
+            DagModel).filter(DagModel.dag_id == self.dag_id).first()
         dp = None
-        if dag and dag.pickle_id:
+
+        if not orm_dag:
+            raise ValueError(
+                "Tried to pickle a DAG that hasn't been loaded yet: {}".format(
+                    self.dag_id))
+
+        if orm_dag.pickle_id:
             dp = session.query(DagPickle).filter(
-                DagPickle.id == dag.pickle_id).first()
+                DagPickle.id == orm_dag.pickle_id).first()
+
         if not dp or dp.pickle != self:
```

Comment:
This line will always be `False` in Python 3 without #315 because comparing two dags (`dp.pickle` and `self`) for equality will always return False. So without #315 calling `dag.pickle()` in Py3 ALWAYS creates a new pickle, even if the DAG was just pickled.